chemreg/substance/serializers.py

```diff
@@ -146,3 +146,32 @@ def validate_score_weight(self, value):
         if not value > 0:
             raise ValidationError("Score Weight must be greater than zero.")
         return value
+
+
+class SynonymSerializer(HyperlinkedModelSerializer):
+    """The serializer for Synonyms."""
+
+    included_serializers = {
+        "source": "chemreg.substance.serializers.SourceSerializer",
+        "substance": "chemreg.substance.serializers.SubstanceSerializer",
+        "synonym_quality": "chemreg.substance.serializers.SynonymQualitySerializer",
+        "synonym_type": "chemreg.substance.serializers.SynonymTypeSerializer",
+    }
+
+    source = ResourceRelatedField(queryset=Source.objects, model=Source)
+    substance = ResourceRelatedField(queryset=Substance.objects, model=Substance)
+    synonym_quality = ResourceRelatedField(
+        queryset=SynonymQuality.objects, model=SynonymQuality
+    )
+    synonym_type = ResourceRelatedField(queryset=SynonymType.objects, model=SynonymType)
```

Comment:
The DRF:JA overrides have a custom way of dealing with related resources.  Rather than having an `ResourceRelatedFields` drive everything the overrides allow you to point a Related Resource directly at its serializer and the includes will be created automatically.

```suggestion
    source = SourceSerializer
    substance = SubstanceSerializer
    synonym_quality = SynonymQualitySerializer
    synonym_type = SynonymTypeSerializer
```

Reply:
Great suggestion!